keystorage/src/main/java/tech/pegasys/web3signer/keystorage/gcp/GcpSecretManager.java

```diff
@@ -0,0 +1,130 @@
+import com.google.cloud.secretmanager.v1.SecretManagerServiceClient;
+import com.google.cloud.secretmanager.v1.SecretPayload;
+import com.google.protobuf.ByteString;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.annotations.NotNull;
+
+public class GcpSecretManager implements Closeable {
+
+  private static final Logger LOG = LogManager.getLogger();
+  private final SecretManagerServiceClient secretManagerServiceClient;
+
+  public GcpSecretManager() throws IOException {
+    secretManagerServiceClient = SecretManagerServiceClient.create();
+  }
+
+  @Override
+  public void close() throws IOException {
+    secretManagerServiceClient.close();
+  }
+
+  /**
+   * Bulk load secrets.
+   *
+   * @param projectId GCP Project Id
+   * @param filter GCP Resource filter
+   * @param mapper The mapper function that can convert secret value to appropriate type
+   * @return SecretValueResult with collection of secret values and error count if any.
+   */
+  public <R> MappedResults<R> mapSecrets(
+      final String projectId,
+      final Optional<String> filter,
+      final BiFunction<String, String, R> mapper) {
+
+    final Set<R> result = ConcurrentHashMap.newKeySet();
+    final AtomicInteger errorCount = new AtomicInteger(0);
+    try {
+      listSecrets(projectId, filter)
+          .forEach(
+              secretEntry -> {
+                try {
+                  final Optional<String> secretValue = fetchStringSecret(secretEntry.getName());
+                  if (secretValue.isEmpty()) {
+                    LOG.warn(
+                        "Failed to fetch secret name '{}', and was discarded",
+                        secretEntry.getName());
+                    errorCount.incrementAndGet();
+                  } else {
+                    MappedResults<R> multiResult =
+                        SecretValueMapperUtil.mapSecretValue(
+                            mapper, secretEntry.getName(), secretValue.get());
+                    result.addAll(multiResult.getValues());
+                    errorCount.addAndGet(multiResult.getErrorCount());
+                  }
+                } catch (final Exception e) {
+                  LOG.warn(
+                      "Failed to map secret '{}' to requested object type due to: {}.",
+                      secretEntry.getName(),
+                      e.getMessage());
+                  errorCount.incrementAndGet();
+                }
+              });
+    } catch (final Exception e) {
+      LOG.warn("Unexpected error during GCP list-secrets operation", e);
+      errorCount.incrementAndGet();
+    }
+    return MappedResults.newInstance(result, errorCount.intValue());
+  }
+
+  private Iterable<Secret> listSecrets(String projectId, Optional<String> filter) {
+    ListSecretsRequest request = listSecretsRequest(projectId, filter);
+    return secretManagerServiceClient.listSecrets(request).iterateAll();
+  }
+
+  @NotNull
```

Comment:
we tend to delete jetbrains' notnull annotation that gets added due to refactoring of the methods.

Reply:
yeah sorry for that, probably would make sense to make the build fail in case of these annotations then ;) I am not sure I did not introduce more of them